bundles/core/org.eclipse.smarthome.core.persistence/src/main/java/org/eclipse/smarthome/core/persistence/rest/PersistenceResource.java

```diff
@@ -0,0 +1,261 @@
+    @ApiResponses(value = @ApiResponse(code = 200, message = "OK") )
+    public Response httpGetPersistenceServices(@Context HttpHeaders headers) {
+        Object responseObject = getPersistenceServiceList();
+        return Response.ok(responseObject).build();
+    }
+
+    @GET
+    @Path("/{servicename: [a-zA-Z_0-9]*}/{itemname: [a-zA-Z_0-9]*}")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Gets item persistence data from the persistence service.", response = ItemHistoryBean.class)
+    @ApiResponses(value = @ApiResponse(code = 200, message = "OK") )
+    public Response httpGetPersistenceItemData(@Context HttpHeaders headers,
+            @PathParam("servicename") String serviceName, @PathParam("itemname") String itemName,
+            @QueryParam("starttime") String startTime, @QueryParam("endtime") String endTime,
+            @QueryParam("page") long pageNumber, @QueryParam("pagelength") long pageLength) {
+
+        final Object responseObject = getItemHistoryBean(serviceName, itemName, startTime, endTime);
+        return Response.ok(responseObject).build();
+    }
+
+    private Date convertTime(String sTime) {
+        DateTimeType dateTime = new DateTimeType(sTime);
+        return dateTime.getCalendar().getTime();
+    }
+
+    private ItemHistoryBean getItemHistoryBean(String serviceName, String itemName, String timeBegin, String timeEnd) {
+        PersistenceService service = persistenceServices.get(serviceName);
+
+        long timerStart = System.currentTimeMillis();
+
+        if (service == null) {
+            logger.debug("Persistence service not found '{}'.", serviceName);
+            throw new WebApplicationException(404);
+        }
+
+        if (!(service instanceof QueryablePersistenceService)) {
+            logger.debug("Persistence service not queryable '{}'.", serviceName);
+            throw new WebApplicationException(404);
+        }
+
+        QueryablePersistenceService qService = (QueryablePersistenceService) service;
+
+        Date dateTimeBegin = new Date();
+        Date dateTimeEnd = dateTimeBegin;
+        if (timeBegin != null) {
+            dateTimeBegin = convertTime(timeBegin);
+        }
+
+        if (timeEnd != null) {
+            dateTimeEnd = convertTime(timeEnd);
+        }
+
+        // End now...
+        if (dateTimeEnd.getTime() == 0) {
+            dateTimeEnd = new Date();
+        }
+        if (dateTimeBegin.getTime() == 0) {
+            dateTimeBegin = new Date(dateTimeEnd.getTime() - 86400000);
+        }
+
+        // Default to 1 days data if the times are the same
+        if (dateTimeBegin.getTime() >= dateTimeEnd.getTime()) {
+            dateTimeBegin = new Date(dateTimeEnd.getTime() - 86400000);
+        }
+
+        FilterCriteria filter = new FilterCriteria();
+        Iterable<HistoricItem> result;
+        State state = null;
+
+        Long quantity = 0l;
+        double average = 0;
```

Comment:
I guess that's also the same for min/max - depending on the persistence service being used (eg rrd doesn't log exact data).
I'm wondering if I should remove these statistical parts of the response - what do you think? HABmin doesn't use it so I don't care and it might be safer to return the data and let the client do whatever it likes to calculate/present this if it wants?